umbral/utils.py

```diff
@@ -55,6 +55,42 @@ def hash_to_bn(list, params):
  
     return res
 
+def hash_to_point(curve, data, constant=None):
+    """
+    Hashes arbitrary data into a valid EC point of the specified curve,
+    using the try-and-increment method.
+    It admits an optional constant as an additional input to the hash function.
+    It uses SHA256 as the internal hash function. 
+
+    WARNING: Do not use when the input data is secret, as this implementation is not 
+    in constant time, and hence, it is not safe with respect to timing attacks.
+
+    TODO: Check how to uniformly generate ycoords. Currently, it only outputs points 
+    where ycoord is even (i.e., starting with 0x02 in compressed notation)
+    """
+    if constant is None:
+        constant = []
+
+    # We use a 32-bit counter as additional input
+    i = 1
+    while i < 2**32:
+        ibytes = i.to_bytes(4, byteorder='big')
+        digest = hashes.Hash(hashes.SHA256(), backend=default_backend())
+        digest.update(constant + ibytes + data)
+        hash = digest.finalize()
+
+        compressed02 = b"\x02"+hash
+        try:
+            h = Point.from_bytes(compressed02, curve)
+            return h
+        except:
```

Comment:
This is not a good approach for this situation.  Which exceptions are we actually catching here?

https://realpython.com/blog/python/the-most-diabolical-python-antipattern/

Reply:
Yeah, if there are ever any exceptions, we'd need to figure out which are those (and catch specifically them if we have to)

Reply:
I believe it throws `cryptography.exceptions.InternalError`

Reply:
And also a potential `ValueError` from `Point.from_bytes`.

Reply:
I agree there must be a better way to do this that I don't know. My intention is 'try to deserialize this potential point, if something fails, then the point is garbage'. I really don't care what's the exception. I just know that if the potential point was valid, then there will be no exception. Feel free to suggest something along this line.

Reply:
Would it not work to use `g` as a scalar for a hashed object?

Essentially:
`u = hash_to_bn(some_data) * g`

There might be some security implications with that... Not sure, need to look into this.

Reply:
> I really don't care what's the exception. 

The thing is: you *do* care about strange or wacky exceptions, like `RuntimeError`, `MemoryError`, or `OSError` - you don't want to accidentally catch those.

Reply:
No, that's not a good approach. We want to give assurance that we don't know the discrete log of u wrt to g.

Reply:
Jmyles, you're right. Didn't think about that.

Reply:
@cygnusv @jMyles that's why it's so important to have both scientists and engineers here!

Reply:
How about just plain ole' humans? ;-)

Reply:
Scientists and engineers are a good mixture :-)

Reply:
Well, let me remind you that I am neither.

I engage in a lot of science, and a lot of engineering, but I'm not more a scientist or engineer than I am a school bus driver.

If you want to appellate y'all selves, that's fine, but I never like being excluded from the pool of people who can tackle a problem because of the perception that I'm in the wrong `class` of human.  I'm down for *any* problem - especially if it's a serious challenge - whether it's science, engineering, or bus driving.

My resume lists my profession as "Chocobo Breeder."

:-)

I will concur, though, that this PR (and also #38 / #24) have brought out of a lot of the most developed strengths in each of the four of us!